packages/graph/hash_graph/lib/graph/src/api/rest/mod.rs

```diff
@@ -52,35 +49,28 @@ fn api_documentation() -> Vec<openapi::OpenApi> {
     ]
 }
 
-async fn read_from_store<'pool, 'id: 'pool, T, P, I, O>(
+async fn read_from_store<'pool, P, T>(
     pool: &'pool P,
-    identifier: I,
-) -> Result<O, StatusCode>
+    query: &<P::Store<'pool> as Read<T>>::Query<'_>,
+) -> Result<Vec<T>, StatusCode>
 where
     P: StorePool,
-    P::Store<'pool>: crud::Read<'id, I, T, Output = O>,
-    I: Send + Debug + Copy + 'id,
+    P::Store<'pool>: Read<T>,
+    T: Send,
 {
     let store = pool.acquire().await.map_err(|report| {
         tracing::error!(error=?report, "Could not acquire access to the store");
         StatusCode::INTERNAL_SERVER_ERROR
     })?;
 
-    crud::Read::<I, T>::get(&store, identifier)
-        .await
-        .map_err(|report| {
-            tracing::error!(error=?report, identifier=?identifier, "Could not read from the store");
-
-            if report.contains::<QueryError>() {
```

Comment:
This test was moved to the caller function. For get-all statements, an empty list is valid, for get-one statements there will be an error on an empty list.